### ISSUE TITLE
crash on Environment(readonly=True).db_open(create=True)

### ISSUE THREAD

**Issue by pspacek** [jnwatson/py-lmdb#160]
### Affected Operating Systems
* Linux, Fedora 25, `python3-3.5.3-6.fc25.x86_64`

### Affected py-lmdb Version
* "print lmdb.__version__": `0.92`
* commit 4651bb3a865c77008ac261443899fe25f88173f2

### py-lmdb Installation Method
* `python3 setup.py install --user`

### Using bundled or distribution-provided LMDB library?
* Bundled

### Distribution name and LMDB library version
* print lmdb.version(): `(0, 9, 19)`

### Describe Your Problem
An (errorneous) attempt to create new database on read-only Environment causes crash:
```
In [10]: import lmdb
In [11]: env = lmdb.Environment(path='env/', create=False, readonly=True, max_dbs=5)

In [12]: env.open_db(b'666', create=False)
---------------------------------------------------------------------------
NotFoundError                             Traceback (most recent call last)
<ipython-input-12-3d6636cec95f> in <module>()
----> 1 env.open_db(b'666', create=False)

NotFoundError: mdb_dbi_open: MDB_NOTFOUND: No matching key/data pair found

In [13]: env.open_db(b'666', create=True)
Segmentation fault (core dumped)
```

### Errors/exceptions Encountered
The Python process crashed.

### Describe What You Expected To Happen
An exception regarding read-only environment should be raised.

### Describe What Happened Instead
The Python process crashed.

**Comment by dw** [comment]
Whoa nice, an actual crash. Haven't seen one in almost 2 years? :P

Will look at this over the course of the week. Thanks,


David

**Comment by pspacek** [comment]
BTW this problem is already covered by test `test_readonly_env_sub_eperm` and the test is crashing on my test system. Weird.

**Comment by pspacek** [comment]
... and the test `test_sub_rotxn` as well. This is super weird.

**Comment by dw** [comment]
Sorry for the delay replying. I reproduced on Debian GNU/Linux 8.7
(jessie) amd64, and hbf fiddled with it for a while. The crash is
definitely in LMDB code, but I'm not convinced the C binding isn't
responsible.

Sorry for lack of time, I've got 3 or 4 commercial projects on just now
:(

**Comment by jnwatson** [comment]
This is a bug upstream and was resolved in 0.9.21. I'll pull that in soon.

**Comment by dw** [comment]
Good luck! Note I had to revert 0.9.19 because it totally wrecked Windows.
Never got around to fixing it, but if you find AppVeyor freaking out when
you land 0.9.21, I can try put in a few hours to look at the linker issue
again. Basically they started depending on DDK symbols and the AppVeyor
machine does not have the DDK installed. There are a bunch of tricks to
work around it, but I couldn't get any of them to work :)
